manticore/__main__.py

```diff
@@ -79,51 +78,38 @@ def parse_arguments():
 def main():
     args = parse_arguments()
 
-    m = Manticore(args.argv[0], args.argv[1:], disasm=args.disasm)
-    m.policy = args.policy
-    m.args = args
+    log.set_verbosity(args.v)
+
+
+    #TODO 
+    initial_state = make_initial_state(args.argv[0], argv=args.argv[1:],  disasm=args.disasm)
+    m = Manticore(initial_state, workspace_url=args.workspace,  policy=args.policy)
+
+    #Fixme remove all this...
+    m.coverage_file = args.coverage
 
     if args.data:
         m.concrete_data = args.data
 
-    if args.workspace:
-        m.workspace = args.workspace
-
     if args.profile:
         m.should_profile = args.profile
 
-    if args.dumpafter != 0:
-        m.dumpafter = args.dumpafter
-
-    if args.maxstorage != 0:
-        m.maxstorage = args.maxstorage
-
-    if args.maxstates != 0:
-        m.maxstates = args.maxstates
-
-    if args.coverage:
-        m.coverage_file = args.coverage
-
     if args.names is not None:
         m.apply_model_hooks(args.names)
 
     if args.env:
```

Comment:
we can move this stuff to the init hook instead of requiring `m.initial_state`?

Reply:
We currently need m.initial_state to publish the event for the init hook.